### DISCUSSION TITLE
Rename master node

### DISCUSSION THREAD

**Discussion by TTAbwill** [k3s-io/k3s#9805] [Q&A]
Hi there,

first of all, thank you for the great project!

I have just changed my home lab cluster from 1 master with sqlite node to 3 masters with etcd and have a comprehension question. 

I would like to install the first (former sqlite) master node nine times and rename it. Does it still have a special role in the cluster, as I used it to join new nodes? Or can I simply delete the node from the cluster, reinstall it and add it back to the cluster with a different hostname?

Thanks in advance.

**Comment by brandond** [comment]
Kubernetes objects, including nodes, cannot be renamed. If you want to rename a node, you should delete it with `kubectl delete node`, stop k3s on that node, change the hostname, and then start it again. It will rejoin with the new name. For servers with embedded etcd, you should delete the etcd datastore files from disk, and ensure that the --server address to points at one of the other servers in the cluster so that it can rejoin the cluster, instead of starting a new separate cluster.

Be careful to ensure you have quorum in the cluster. For 3 nodes, that would mean having at least 2 nodes running at all atimes.

**Reply by antonengelhardt** [reply]
Hi, do you know by any chance what happens to volumes stored on the node that i want to rename? Im talking about volumes that are using local path storage class. Thx
